contracts/BNUGToken.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity >=0.4.22 <0.9.0;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+
+
+contract BNUGToken is ERC20Burnable {
+  address contractCreator;
+  uint private _totalSupply = 200000000 * 10 ** 18;
+  mapping(address => uint) private balances;
+  mapping(address => mapping(address => uint)) allowed;
+
+  modifier onlyCreator {
+    require(msg.sender == contractCreator, "Only the contract creator can execute this function");
+    _;
+  }
+
+  event Burn(uint indexed burned);
+
+  constructor(address creator) public ERC20("Blockchain Nigeria User Group", "BNUG") {
```

Comment:
BNUGToken contract should only implement a constructor with an internal mint function for total supply